fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -191,17 +191,30 @@ def fit(self, X, y, **kwargs):
                      len(lagrangian.predictors))
 
     def predict(self, X):
-        """Provide a prediction for the given input data.
-
-        Note that this is non-deterministic, due to the nature of the
-        exponentiated gradient algorithm.
-
-        :param X: Feature data
-        :type X: numpy.ndarray or pandas.DataFrame
-
-        :return: The prediction. If `X` represents the data for a single example
+        """Provide predictions for the given input data.
+
+        Predictions are randomized, i.e., repeatedly calling `predict` with
+        the same feature data may yield different output. This
+        non-deterministic behavior is intended and stems from the nature of
+        the exponentiated gradient algorithm.
+
+        Notes
+        -----
+        A fitted `ExponentiatedGradient` has the members `predictors_` and
+        `weights_`. On each data point in `X` it picks one of its underlying
+        predictors randomly and applies it. `weights_` determines the
+        probabilities to choose the predictors.
```

Comment:
A minor rewrite for improved clarity and also, a small thing, I think that standard terminology is that Python class instances have `attributes`:

> A fitted `ExponentiatedGradient` has an attribute `predictors_`, an
> array of predictors, and an attribute `weights_`, an array of non-negative floats of the same length.
> The prediction on each data point in `X` is obtained by first picking a
> random predictor according to the probabilities in `weights_` and then
> applying it. Different predictors can be chosen on different data points.